views/js/previewer/plugins/content/scoringHighlighter.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Highlighter Plugin
+ */
+define([
+    'jquery',
+    'i18n',
+    'ui/hider',
+    'taoTests/runner/plugin',
+    'tpl!ltiOutcomeUi/previewer/plugins/content/tpl/highlighter-tray',
+    'ui/highlighter',
+    'css!ltiOutcomeUi/previewer/plugins/content/css/highlighterTray.css'
+], function ($, __, hider, pluginFactory, highlighterTrayTpl, highlighterFactory) {
+    'use strict';
+
+    function highlight(highlighter, selection) {
+        highlighter.highlightRanges(getAllRanges(selection));
+        //Sending the highlighIndex to parent so that it can be saved on MS side
+        selection.removeAllRanges();
+        window.parent.postMessage({ event: 'indexUpdated', payload: highlighter.getHighlightIndex() }, '*');
+    }
+
+    function clearHighlights(highlighter) {
+        highlighter.clearHighlights();
+        window.parent.postMessage({ event: 'indexUpdated', payload: highlighter.getHighlightIndex() }, '*');
+    }
+
+    function getAllRanges(selection) {
+        const allRanges = [];
+
+        for (let i = 0; i < selection.rangeCount; i++) {
+            allRanges.push(selection.getRangeAt(i));
+        }
+        return allRanges;
+    }
+
+    return pluginFactory({
+        name: 'highlighter',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init() {
+            const testRunner = this.getTestRunner();
+
+            if (!window.getSelection) {
```

Comment:
Is it necessary? Seems it won't be used in ie8. And be more careful with throwing errors:
1. Make sure that it won't break other features
2. JSDoc it